lib/_http_client.js

```diff
@@ -325,13 +357,21 @@ ClientRequest.prototype.abort = function abort() {
   // the request queue through handling in onSocket.
   if (this.socket) {
     // in-progress
-    this.socket.destroy();
+    this.socket.destroy(err, cb);
+  } else if (this.upgradeOrConnect) {
+    // We're not waiting for a socket.
+    cb(err);
+  } else {
+    this[kWritableState].destroyCallback = er => cb(er || err);
```

Comment:
@mcollina: What should be priority here? Error passed into `destroy(err)` (i.e. `err`) or error from `socket.destroy(null, cb)` (i.e. `er`)?